cli/ops/tty.rs

```diff
@@ -250,3 +251,113 @@ pub fn op_isatty(
     })?;
   Ok(JsonOp::Sync(json!(isatty)))
 }
+
+#[derive(Serialize)]
+struct ConsoleSize {
+  columns: u32,
+  rows: u32,
+}
+
+#[cfg(unix)]
+fn console_size(rid: i32) -> Result<ConsoleSize, std::io::Error> {
+  use libc::{ioctl, winsize, TIOCGWINSZ};
+  use std::mem::MaybeUninit;
+
+  let mut count: u32 = 0;
+  loop {
+    count += 1;
+
+    let mut buffer = MaybeUninit::<winsize>::uninit();
+    let result =
+      unsafe { ioctl(rid, TIOCGWINSZ, buffer.as_mut_ptr() as *mut _) };
+
+    if result == -1 {
+      if count > 3 {
+        break unsafe { MaybeUninit::zeroed().assume_init() };
+      }
+      continue;
+    } else {
+      let console_data = unsafe { buffer.assume_init() };
+      break Ok(ConsoleSize {
+        columns: console_data.ws_col as u32,
+        rows: console_data.ws_row as u32,
+      });
+    }
+  }
+}
+
+#[cfg(windows)]
+fn console_size(_rid: i32) -> Result<ConsoleSize, std::io::Error> {
+  use winapi::um::winbase::STD_OUTPUT_HANDLE;
+  use winapi::um::wincon::GetConsoleScreenBufferInfo;
+  use winapi::um::wincon::{CONSOLE_SCREEN_BUFFER_INFO, COORD, SMALL_RECT};
+
+  let null_coord = COORD { X: 0, Y: 0 };
+  let null_smallrect = SMALL_RECT {
+    Left: 0,
+    Top: 0,
+    Right: 0,
+    Bottom: 0,
+  };
+
+  let mut console_data = CONSOLE_SCREEN_BUFFER_INFO {
+    dwSize: null_coord,
+    dwCursorPosition: null_coord,
+    wAttributes: 0,
+    srWindow: null_smallrect,
+    dwMaximumWindowSize: null_coord,
+  };
+
+  // let mut resource_table = isolate_state.resource_table.borrow_mut();
+  // let _result = std_file_resource(
+  //   &mut resource_table,
+  //   args.rid as u32,
+  //   move |r| match r {
+  //     Ok(std_file) => {
+  //       let handle = get_windows_handle(&std_file)
+  //         .expect("Could not get windows handle.");
+  //
+  //       // For testing purposes
+  //       assert!(
+  //         unsafe { winapi::um::processenv::GetStdHandle(STD_OUTPUT_HANDLE) }
+  //           == handle
+  //       );
+  //
+  //       Ok(true)
+  //     }
+  //     _ => Ok(false),
+  //   },
+  // )
+  // .expect("Could not get console screen buffer info.");
```

Comment:
Dead code should be removed before landing

Reply:
I left this code in reference to my last comment to show my previous implementation; I will remove it before merging.